tmk_core/common/backlight.h

```diff
@@ -20,6 +20,12 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <stdint.h>
 #include <stdbool.h>
 
+#ifndef BACKLIGHT_LEVELS
+  #define BACKLIGHT_LEVELS 3
+#elif BACKLIGHT_LEVELS > 15
+  #error "Maximum value of BACKLIGHT_LEVELS is 15"
```

Comment:
Is there any way to "fail" this gracefully, by perhaps resetting the value on the fly? Maybe a warning that the user's defined setting is too high, and then reset their value to 15?

Reply:
I think just MOD'ing the value would fix this. It the setter simply apply % 15 to the incoming value to make it roll over. That's the safest thing to do. If you want to get fancy you could also scale the value to make sure it's always a fraction of 15, but that would require doing some math. e.g. if the max is 255, then incoming * 0.39 should make it safe.

Reply:
also, `#pragma message` would work.

Reply:
I'm trying to go with least astonishment here, so I don't really want to modify the value in any way, just tell the user that it's invalid.

Now that I've limited the level in EEPROM to 4 bits, IMO values above 15 should either be outright rejected, or the user politely told that they may get some funky backlight behaviour. For example, `backlight_increase()` will only stop when it reaches `BACKLIGHT_LEVELS`. Assuming `backlight_config.level++` wraps around at 15, it will never reach it and instead act like `BL_STEP`.

Reply:
@skullydazed @jackhumbert ?